### PR TITLE
added support for building with the GNU Autotools

### PR DESCRIPTION
**Pull request by marcomaggi**

### CODE REVIEW

**Comment by uael** [comment]
Awesome one, I'm loving autotools even if it's not cross platform :+1: 
Can you and some builds in the travis matrix using it ?

File: README.autotools [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+See the LICENSE file.
+
+
+3. Install
+----------
+
+To install from a proper release tarball, do this:
+
+    $ cd libsv-3.1
```
Comment (uael) [inline README.autotools:41]: 3.1 ?

**Comment by marcomaggi** [comment]
I'm a real beginner with Travis; I will try something and update the pull request.

File: .travis.yml [hunk 2]
```diff
@@ -96,6 +96,30 @@ matrix:
...
+	  - autotools.marcomaggi-2017-05-17
+      compiler: gcc-5
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - gcc-5
+      install:
+        - ./meta/travis-ci/install-autoconf.sh
```
Comment (uael) [inline .travis.yml]: Travis already have autotools and automake installed :smiley:

File: .travis.yml [hunk 3]
```diff
@@ -96,6 +96,30 @@ matrix:
...
+	- ./meta/travis-ci/install-automake.sh
+	- ./meta/travis-ci/install-texinfo.sh
+	- ./meta/travis-ci/install-pkg-config.sh
+      script: |
+        export PATH=/tmp/mine/bin:$PATH;
+	export LD_LIBRARY_PATH=/tmp/mine/lib:$LD_LIBRARY_PATH;
+	export PKG_CONFIG_PATH=/tmp/mine/lib/pkgconfig:$PKG_CONFIG_PATH;
+  	sh ./autogen.sh;
+        ./configure --enable-maintainer-mode CC=/usr/bin/gcc-5;
```
Comment (uael) [inline .travis.yml]: CC is already defined by travis, see line 103

**Comment by uael** [comment]
@marcomaggi [https://docs.travis-ci.com/user/languages/c/](https://docs.travis-ci.com/user/languages/c/)
Thank you!

File: README.autotools (hunk 1 above)
Comment (marcomaggi) [reply README.autotools:41]: I got that version number from a CMake file; I do not know CMake, so it was  a mistake.  What is the version number then?

File: .travis.yml (hunk 2 above)
Comment (marcomaggi) [reply .travis.yml]: Not the right version, at least not in the old distributions they use.  I want the latest releases, which are some years old.

File: .travis.yml (hunk 3 above)
Comment (marcomaggi) [reply .travis.yml]: Fine then.  I will unset it.

File: .travis.yml [hunk 4]
```diff
@@ -117,7 +117,7 @@ matrix:
 	export LD_LIBRARY_PATH=/tmp/mine/lib:$LD_LIBRARY_PATH;
```
Comment (uael) [inline .travis.yml]: You should also use spaces instead tab

File: README.autotools (hunk 1 above)
Comment (uael) [reply README.autotools:41]: The first release will be 1.0.0 until it's still 0.0.1

File: .travis.yml (hunk 2 above)
Comment (uael) [reply .travis.yml]: You can use the trusty dist `dist: trusty` it will accelerate the build time and avoid the need of such scripts

**Review by uael — APPROVED** [review]

**Review by uael — APPROVED** [review]

**Pull request merged**